frontend/src/app/forms/side-menu/side-menu.component.html

```diff
@@ -9,32 +9,77 @@ <h1>Progress</h1>
         </div>
       </div>
 
-      <ion-list lines="none" *ngFor="let menu of sideMenuList">
-        <ion-menu-toggle auto-hide="true">
+      <ion-list lines="none">
+        <ng-container
+          [ngTemplateOutlet]="menuItem"
+          [ngTemplateOutletContext]="{
+            step: APPLICATION_STEP.BasicInformation,
+            displayName: 'Basic Information',
+            isValid: formsData.controls.basicInfo.valid,
+            isActive: currentStep === APPLICATION_STEP.BasicInformation
+          }"
+        ></ng-container>
+
+        <ng-container
+          [ngTemplateOutlet]="menuItem"
+          [ngTemplateOutletContext]="{
+            step: APPLICATION_STEP.ContactInformation,
+            displayName: 'Contact Information',
+            isValid: formsData.controls.contactInfo.valid,
+            isActive: currentStep === APPLICATION_STEP.ContactInformation
+          }"
+        ></ng-container>
+
+        <ion-menu-toggle>
           <ion-item
```

Comment:
Could have been added in ng-container as only the icon name varies which can be handled in the ternary check.

Reply:
The ng-template also has properties like "isValid" that aren't really useful here, so I felt that adding a separate item allows us to later change/style it later much easier. But we can also move it into the ng-template - I don't really have a strong opinion about that.